services/src/pce/src/main/java/org/openkilda/pce/provider/NeoDriver.java

```diff
@@ -235,26 +237,48 @@ public NeoDriver(Driver driver) {
                         + "sw.state as state "
                         + "order by sw.name";
         logger.debug("Executing getSwitches Query: {}", q);
-
-        List<SwitchInfoData> results = new LinkedList<>();
         try (Session session = driver.session(AccessMode.READ)) {
             StatementResult queryResults = session.run(q);
+            List<SwitchInfoData> results = new LinkedList<>();
             for (Record record : queryResults.list()) {
-                SwitchInfoData sw = new SwitchInfoData();
-                sw.setAddress(record.get("address").asString());
-                sw.setController(record.get("controller").asString());
-                sw.setDescription(record.get("description").asString());
-                sw.setHostname(record.get("hostname").asString());
-
-                String status = record.get("state").asString();
-                SwitchState st = ("active".equals(status)) ? SwitchState.ACTIVATED : SwitchState.CACHED;
-                sw.setState(st);
-
-                sw.setSwitchId(new SwitchId(record.get("name").asString()));
-                results.add(sw);
+                results.add(buildSwitchInfoData(record));
             }
+            return results;
         }
-        return results;
+    }
+
+    @Override
+    public Optional<SwitchInfoData> getSwitchById(SwitchId id) {
+        String q = "MATCH (sw:switch) "
+                + "WHERE sw.name = {switchId} "
+                + "RETURN "
+                + "sw.name as name, "
```

Comment:
We can return whole "sw" i.e. `RETURN sw`. In this case we avoid mandatory copy paste of "return" part(it must be same because of result parsing logic). The cost is one "null" check during result parsing.